p2p/__init__.py

```diff
@@ -1413,70 +1432,77 @@ def _check_for_errors(self, resp, req_url):
-                elif u'unknown attribute' in resp.content:
+                elif u'unknown attribute' in resp_content:
                     raise P2PUnknownAttribute(resp.url, request_log, curl)
-                elif u"Invalid access definition" in resp.content:
+                elif u"Invalid access definition" in resp_content:
                     raise P2PInvalidAccessDefinition(resp.url, request_log, \
 curl)
-                elif u"solr.tila.trb" in resp.content:
+                elif u"solr.tila.trb" in resp_content:
                     raise P2PSearchError(resp.url, request_log, curl)
-                elif u"Request Timeout" in resp.content:
+                elif u"Request Timeout" in resp_content:
                     raise P2PTimeoutError(resp.url, request_log, curl)
-                elif u'Duplicate entry' in resp.content:
+                elif u'Duplicate entry' in resp_content:
                     raise P2PUniqueConstraintViolated(resp.url, request_log, \
 curl)
                 elif (u'Failed to upload image to the photo service'
-                        in resp.content):
+                        in resp_content):
                     raise P2PPhotoUploadError(resp.url, request_log, curl)
-                elif u"This file type is not supported" in resp.content:
+                elif u"This file type is not supported" in resp_content:
                     raise P2PInvalidFileType(resp.url, request_log, curl)
-                elif re.search(r"The URL (.*) does not exist", resp.content):
+                elif re.search(r"The URL (.*) does not exist", resp_content):
                     raise P2PFileURLNotFound(resp.url, request_log)
 
                 data = resp.json()
 
-            except ValueError:
+            except (ValueError, TypeError):
                 pass
             raise P2PException(resp.url, request_log, curl)
         elif resp.status_code == 404:
             raise P2PNotFound(resp.url, request_log, curl)
         elif resp.status_code >= 400:
-            if u'{"slug":["has already been taken"]}' in resp.content:
+            if u'{"slug":["has already been taken"]}' in resp_content:
                 raise P2PSlugTaken(resp.url, request_log, curl)
-            elif u'{"code":["has already been taken"]}' in resp.content:
+            elif u'{"code":["has already been taken"]}' in resp_content:
                 raise P2PSlugTaken(resp.url, request_log, curl)
             elif resp.status_code == 403:
                 raise P2PForbidden(resp.url, request_log, curl)
             try:
                 resp.json()
             except ValueError:
                 pass
-            raise P2PException(resp.content, request_log, curl)
+            raise P2PException(resp_content, request_log, curl)
         return request_log
 
+    def convert_response_bytes_to_string(self, response):
+        if str(type(response.content)) == "<class 'bytes'>":
+            return response.content.decode("utf-8") 
```

Comment:
If this test comes back as false, does the function return nothing?

Reply:
I think it will return `None`. Probably better to be explicit about that. Would ideal behavior be to return `None` or an empty string? I'm thinking empty string so that `_check_for_errors` doesn't fail when looking for substrings

Reply:
Am I nuts, or will that cause an AttributeError issue [here](https://github.com/datadesk/p2p-python/pull/47/files/fcf6c1cc202210e46f1ea113573b24a1018feb69#diff-287cde72acc44da4d5c07108a734f1d4R1452). I've only read the code, not run it, so forgive me if I'm being stupid. 

```
resp_content = self.convert_response_bytes_to_string(resp)

if resp.status_code >= 500:
```

Reply:
Updated with additional check for `str` type and fallback to an empty string if `response.content` is not of a string type.